pkg/backup/backup.go

```diff
@@ -167,101 +174,128 @@ func (b *Backup) Run() {
 		if ackchan != nil {
 			ack := backupNowAck{err: err}
 			if err == nil {
-				ack.status = b.getLatestBackupStatus()
+				ack.status = bc.backupManager.getLatestBackupStatus()
 			}
 			ackchan <- ack
 		}
 	}
 }
 
-func (b *Backup) saveSnap(lastSnapRev int64) (int64, error) {
-	podList, err := b.kclient.Core().Pods(b.namespace).List(k8sutil.ClusterListOpt(b.clusterName))
+// saveSnap saves the latest snapshot if latest revision is greater than the given lastSnapRev.
+func (b *BackupManager) saveSnap(lastSnapRev int64) (int64, error) {
+	start := time.Now()
+
+	n, rev, etcdVersion, err := SaveSnap(b.kclient, b.etcdTLSConfig, lastSnapRev, b.be, b.namespace, b.clusterName)
 	if err != nil {
-		return lastSnapRev, err
+		return lastSnapRev, fmt.Errorf("failed to SaveSnap: (%v)", err)
 	}
 
-	var pods []*v1.Pod
-	for i := range podList.Items {
-		pod := &podList.Items[i]
-		if pod.Status.Phase == v1.PodRunning {
-			pods = append(pods, pod)
-		}
+	bs := backupapi.BackupStatus{
+		CreationTime:     time.Now().Format(time.RFC3339),
+		Size:             util.ToMB(n),
+		Version:          etcdVersion,
+		Revision:         rev,
+		TimeTookInSecond: int(time.Since(start).Seconds() + 1),
 	}
+	b.recentBackupsStatus = append(b.recentBackupsStatus, bs)
+	if len(b.recentBackupsStatus) > maxRecentBackupStatusCount {
+		b.recentBackupsStatus = b.recentBackupsStatus[1:]
+	}
+
+	return rev, nil
+}
 
+// SaveSnap saves snapshot to backend from the etcd member whose rev is greater than lastSnapRev.
+func SaveSnap(kclient kubernetes.Interface, tls *tls.Config, lastSnapRev int64, be backend.Backend, namespace, clusterName string) (int64, int64, string, error) {
```

Comment:
why do we need to create this func there? we should expose backupManager and its methods.

Reply:
/cc @fanminshi this needs to be resolved.

Reply:
kk, let me see what I can do.